### DISCUSSION TITLE
Allow web cam streams with MediaElement

### DISCUSSION THREAD

**Discussion by AndreasReitberger** [CommunityToolkit/Maui#936] [New Feature Discussions]
I'm migrating form Xamarin.Forms, where I'm using LibVlc to stream from a webcam, which is working fine. Since the LibVlcPlayer isn't migrated to MAUI yet, I'm searching for other solutions. Best pick would be a maui component like the `MediaElement`.

The `Uri` for the stream looks something like that.

```cs
[ObservableProperty]
string uri = "http://192.168.10.112/printer/cammjpg/my_printer?cam=1&apikey=yy";
```

Using `LibVlc` the implemention in XF looks like following:

```cs
string uri = "http://192.168.10.112/printer/cammjpg/my_printer?cam=1&apikey=yy";

using Media Media = new(GlobalLibVlcManager.Instance.LibVLC, uri, FromType.FromLocation);

Media.AddOption(string.Format(":network-caching={0}", NetworkBufferTime));
Media.AddOption(string.Format(":clock-jitter={0}", 0));
Media.AddOption(string.Format(":clock-synchro={0}", 0));

MediaParsedStatus result = await Media.Parse(MediaParseOptions.ParseNetwork);

MediaPlayer = new MediaPlayer(Media)
{
    EnableHardwareDecoding = true,
    NetworkCaching = Convert.ToUInt32(NetworkBufferTime),
    FileCaching = Convert.ToUInt32(FileCaching),
};
```

Is this something the `MediaElement` is capable of? At the moment I get an error if I use such an uri as `Source`

![image](https://user-images.githubusercontent.com/2952443/215106102-95b69149-06ce-4bcb-bb9a-80d00001c9e9.png)

**Comment by astromediaonly** [comment]
I would love to find a way to play jpeg video streams from url too and MediaElement looked like good component for it, hopefully it can be added?

**Comment by GeorgeLeithead** [comment]
This is already under discussion in #1017

**Reply by astromediaonly** [reply]
The discussion seems to be iOS specific but this thread was not specified and if I remember correctly I did run into this while trying to develop Windows app (with possibility to have Android version later). So if its supposed to work atm anywhere except iOS let me know and I will investigate where I did messed up I guess :)

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889 where it was agreed that New Features need to get above 7 upvotes within 6 months to progress to being reviewed by the maintainers.
